website/content/v1.6/advanced/customizing-the-kernel.md

```diff
@@ -5,50 +5,91 @@ aliases:
+The kernel configuration is located in the files `kernel/build/config-ARCH` files.
+It can be modified using the text editor, or by using the Linux kernel `menuconfig` tool:
 
-To build the image, run:
+```shell
+make kernel-menuconfig
+```
+
+The kernel configuration can be cleaned up by running:
+
+```shell
+make kernel-olddefconfig
+```
+
+Both commands will output the new configuration to the `kernel/build/config-ARCH` files.
+
+Once ready, build the kernel any out-of-tree modules (if required, e.g. `zfs`) and push the artifacts to a registry:
 
-```bash
-DOCKER_BUILDKIT=0 docker build --build-arg RM="/lib/modules" -t installer:kernel .
+```shell
+make kernel REGISTRY=127.0.0.1:5005 PUSH=true
 ```
 
-> Note: buildkit has a bug [#816](https://github.com/moby/buildkit/issues/816), to disable it use `DOCKER_BUILDKIT=0`
+By default, this command will compile and push the kernel both for `amd64` and `arm64` architectures, but you can specify a single architecture by overriding
+a variable `PLATFORM`:
+
+```shell
+make kernel REGISTRY=127.0.0.1:5005 PUSH=true PLATFORM=linux/amd64
+```
+
+This will create a container image `127.0.0.1:5005/siderolabs/kernel:$TAG` with the kernel and modules.
+
+## Building Talos Base Artifacts
+
+Follow the [Building Custom Images]({{< relref "building-images" >}}) guide to set up the Talos source code checkout.
+
+If some new kernel modules were introduced, adjust the list of the default modules compiled into the Talos `initramfs` by
+editing the file `hack/modules-ARCH.txt`.
+
+Try building base Talos artifacts:
+
+```shell
+make kernel initramfs PKG_KERNEL=127.0.0.1:5005/siderolabs/kernel:$TAG PLATFORM=linux/amd64
+```
+
+This should create a new image of the kernel and initramfs in `_out/vmlinuz-amd64` and `_out/initramfs-amd64.xz` respectively.
+
+> Note: if building for `arm64`, replace `amd64` with `arm64` in the commands above.
+
+As a final step, produce the new `imager` container image which can generate Talos boot assets:
+
+```shell
+make imager PKG_KERNEL=127.0.0.1:5005/siderolabs/kernel:$TAG PLATFORM=linux/amd64 INSTALLER_ARCH=targetarch
+```
+
+> Note: if you built the kernel for both `amd64` and `arm64`, a multi-arch `imager` container can be built as well by specifying `INSTALLER_ARCH=all` and `PLATFORM=linux/amd64,linux/arm64`.
+
+## Building Talos Boot Assets
+
+Follow the [Boot Assets]({{< relref "../talos-guides/install/boot-assets" >}}) guide to build Talos boot assets you might need to boot Talos: ISO, `installer` image, etc.
+Replace the reference to the `imager` in guide with the reference to the `imager` container built above.
 
-Now that we have a custom installer we can build Talos for the specific platform we wish to deploy to.
+> Note: if you update the `imager` container, don't forget to `docker pull` it, as `docker` caches pulled images and won't pull the updated image automatically.
```

Comment:
we should update any reference to `docker run` with `docker run --pull=akways` makes it easier

Reply:
I believe it was added relatively recently, so not all users might have it yet: https://github.com/docker/cli/pull/1498